node/src/database/rocksdb.rs

```diff
@@ -217,10 +273,151 @@ impl<'db, DB: DBAccess> Tx for Transaction<'db, DB> {
+
+        match data {
+            // None has a meaning key not found
+            None => Ok(None),
+            Some(blob) => {
+                Ok(Some(ledger::Transaction::read(&mut &blob.to_vec()[..])?))
+            }
+        }
+    }
+
+    fn get_tx_exists(&self, h: [u8; 32]) -> bool {
+        // Check for hash if exists without deserializing
+        self.snapshot.get_cf(self.mempool_cf, h).is_ok()
+    }
+
+    fn delete_tx(&self, h: [u8; 32]) -> Result<bool> {
+        let tx = self.get_tx(h)?;
+        if let Some(tx) = tx {
+            let hash = tx.hash();
+
+            self.inner.delete_cf(self.mempool_cf, hash)?;
+
+            // Delete Secondary indexes
+            // Delete Nullifiers
+            for n in tx.inner.inputs().into_iter() {
+                let key: [u8; 32] = n.to_bytes().into();
+                self.inner.delete_cf(self.nullifiers_cf, key)?;
+            }
+
+            // Delete Fee_Hash
+            self.inner.delete_cf(
+                self.fees_cf,
+                serialize_fee_key(tx.gas_price(), hash)?,
+            )?;
+
+            return Ok(true);
+        }
+
+        Ok(false)
+    }
+
+    fn get_any_nullifier_exists(&self, nullifiers: Vec<[u8; 32]>) -> bool {
+        nullifiers
+            .into_iter()
+            .all(|n| self.snapshot.get_cf(self.nullifiers_cf, n).is_ok())
+    }
+
+    fn get_txs_sorted_by_fee(
+        &self,
+        max_gas_limit: u64,
+    ) -> Result<Vec<Option<ledger::Transaction>>> {
+        // The slippageGasLimit is the threshold that consider the "estimated
+        // gas spent" acceptable even if it exceeds the strict GasLimit.
+        // This is required to avoid to iterate the whole mempool until
+        // it fit perfectly the block GasLimit
+        let slippage_gas_limit = max_gas_limit + max_gas_limit / 10;
+
+        let mut iter = self.inner.raw_iterator_cf(self.fees_cf);
+        iter.seek_to_last();
+
+        let mut total_gas: u64 = 0;
+        let mut txs_list = vec![];
+
+        // Iterate all keys from the end in reverse lexicographic order
+        while iter.valid() {
+            if let Some(key) = iter.key() {
+                let (read_gp, tx_hash) =
+                    deserialize_fee_key(&mut &key.to_vec()[..])?;
+
+                let mut tx = self.get_tx(tx_hash)?;
+                if let Some(tx) = tx {
+                    let gas_price = tx.gas_price();
+                    debug_assert_eq!(read_gp, gas_price);
+                    total_gas += gas_price;
+
+                    if total_gas > slippage_gas_limit {
+                        break;
+                    }
+
+                    txs_list.push(Some(tx));
+                }
+            }
+
+            iter.prev();
+        }
+
+        Ok(txs_list)
+    }
+}
+
+fn serialize_fee_key(fee: u64, hash: [u8; 32]) -> std::io::Result<Vec<u8>> {
+    let mut w = vec![];
+    std::io::Write::write_all(&mut w, &fee.to_be_bytes())?;
```

Comment:
Any specific reason to use BigEndian? IIRC we use LittleEndian across all our stake

Reply:
Yes, RocksDB keys are sorted in lexicographic order so the MSB in multi-byte integers should be the first one.